File: lc/py/067.py
class Solution:
    def isSubtree(self, s: TreeNode, t: TreeNode) -> bool:
        def sameTree(s, t):
            if not s and not t:
                return True
            elif not s or not t:
                return False
            elif s.val != t.val:
                return False
            return sameTree(s.left, t.left) and sameTree(s.right, t.right)

        def dfs(s):
            if not s:
                return False
            elif s.val == t.val and sameTree(s, t):
                return True
            return dfs(s.left) or dfs(s.right)

        return dfs(s)

File: lc/c/067.c
bool sameTree(struct TreeNode *s, struct TreeNode *t) {
    if (!s && !t)
        return true;
    else if (!s || !t)
        return false;
    else if (s->val != t->val)
        return false;
    return sameTree(s->left, t->left) && sameTree(s->right, t->right);
}

bool isSubtree(struct TreeNode* s, struct TreeNode* t){
    if (!s)
        return false;
    else if (s->val == t->val && sameTree(s, t))
        return true;
    return isSubtree(s->left, t) || isSubtree(s->right, t);
}
